test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +91,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (new Parser().parseCommand(inputCommand).isMutating()) {
```

Comment:
How about the else branch? How to test that there is no mutating when a command is not mutating? A good test will cover both cases.